packages/core/src/utils/AssetUtil.ts

```diff
@@ -1,7 +1,25 @@
+import { ApiController } from '../controllers/ApiController.js'
 import { AssetController } from '../controllers/AssetController.js'
 import type { CaipNetwork, Connector, WcWallet } from './TypeUtil.js'
 
 export const AssetUtil = {
+  async fetchWalletImage(imageId?: string) {
+    if (imageId) {
+      await ApiController._fetchWalletImage(imageId)
+      return this.getWalletImageById(imageId)
+    }
+
+    return undefined
+  },
+
+  getWalletImageById(imageId?: string) {
+    if (imageId) {
+      return AssetController.state.walletImages[imageId]
+    }
+
+    return undefined
```

Comment:
```suggestion

    return AssetController.state?.walletImages?.[imageId]
```

Reply:
The `image_id` is optional field in the wallet object, there is always a case that it could be undefined, thus it's necessary to check if the imageId is valid to make a property access